pkg/controller/statusmanager/status_manager.go

```diff
@@ -53,72 +55,73 @@ func New(client client.Client, name, version string) *StatusManager {
-				{Name: "operator", Version: releaseVersion},
+		if reachedAvailableLevel {
+			if releaseVersion := os.Getenv("RELEASE_VERSION"); len(releaseVersion) > 0 {
+				co.Status.Versions = []configv1.OperandVersion{
+					{Name: "operator", Version: releaseVersion},
+				}
+			} else {
+				co.Status.Versions = nil
 			}
-		} else {
-			co.Status.Versions = nil
 		}
-	}
-	for _, condition := range conditions {
-		v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
-	}
+		for _, condition := range conditions {
+			v1helpers.SetStatusCondition(&co.Status.Conditions, condition)
+		}
+
+		progressingCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorProgressing)
+		availableCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorAvailable)
+		if availableCondition == nil && progressingCondition != nil && progressingCondition.Status == configv1.ConditionTrue {
+			v1helpers.SetStatusCondition(&co.Status.Conditions,
+				configv1.ClusterOperatorStatusCondition{
+					Type:    configv1.OperatorAvailable,
+					Status:  configv1.ConditionFalse,
+					Reason:  "Startup",
+					Message: "The network is starting up",
+				},
+			)
+		}
 
-	progressingCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorProgressing)
-	availableCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorAvailable)
-	if availableCondition == nil && progressingCondition != nil && progressingCondition.Status == configv1.ConditionTrue {
 		v1helpers.SetStatusCondition(&co.Status.Conditions,
 			configv1.ClusterOperatorStatusCondition{
-				Type:    configv1.OperatorAvailable,
-				Status:  configv1.ConditionFalse,
-				Reason:  "Startup",
-				Message: "The network is starting up",
+				Type:   configv1.OperatorUpgradeable,
+				Status: configv1.ConditionTrue,
 			},
 		)
-	}
 
-	v1helpers.SetStatusCondition(&co.Status.Conditions,
-		configv1.ClusterOperatorStatusCondition{
-			Type:   configv1.OperatorUpgradeable,
-			Status: configv1.ConditionTrue,
-		},
-	)
-
-	if reflect.DeepEqual(*oldStatus, co.Status) {
-		return
-	}
+		if reflect.DeepEqual(*oldStatus, co.Status) {
+			return nil
+		}
 
-	buf, err := yaml.Marshal(co.Status.Conditions)
-	if err != nil {
-		buf = []byte(fmt.Sprintf("(failed to convert to YAML: %s)", err))
-	}
-	if isNotFound {
-		if err := status.client.Create(context.TODO(), co); err != nil {
-			log.Printf("Failed to create ClusterOperator %q: %v", co.Name, err)
-		} else {
-			log.Printf("Created ClusterOperator with conditions:\n%s", string(buf))
+		buf, err := yaml.Marshal(co.Status.Conditions)
+		if err != nil {
+			buf = []byte(fmt.Sprintf("(failed to convert to YAML: %s)", err))
+		}
+		if isNotFound {
+			if err := status.client.Create(context.TODO(), co); err != nil {
+				return fmt.Errorf("Failed to create ClusterOperator %q: %v", co.Name, err)
+			}
+			return fmt.Errorf("Created ClusterOperator with conditions:\n%s", string(buf))
```

Comment:
er, no, don't return an error on success. Figure out another way to handle the logging